From 5650e7d15d9a267d361773b2357307836dc5888c Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 09:22:49 +0000
Subject: [PATCH] [wasm][test] Force static library to avoid unsupported shared
 library build

Linking libraries without `-static` flag produces unexpected main
function reference error. This is because the linker is trying to link
input objects as executable, not as library.
---
 test/lit.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/lit.cfg b/test/lit.cfg
index c2931990af7..89b8137012a 100644
--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -1859,7 +1859,7 @@ elif run_os == 'wasi':
         config.swift_driver_test_options, swift_execution_tests_extra_flags])
     config.target_codesign = "echo"
     config.target_build_swift_dylib = (
-        "%s -parse-as-library -emit-library -o '\\1'"
+        "%s -parse-as-library -emit-library -static -o '\\1'"
         % (config.target_build_swift))
     config.target_add_rpath = ''
     config.target_swift_frontend = ' '.join([
-- 
2.40.0

